File: keiner_lindarte_aro/instructor_henry/mis_programas/ruby/orientado_a_objetos/3_clase_de_02_agosto/actividad.rb
class Calculadora
  attr_accessor :numb1, :numb2, :operacion

  def initialize(n1, n2, operacion)

    self.numb1 = n1
    self.numb2 = n2
    self.operacion = operacion
    
  end

  def suma()
    return (numb1 + numb2)
  end
  
  def resta()
    return (numb1 - numb2)
  end
  
  def multiplicacion()
    return (numb1 * numb2)
  end
  
  def division()
    return (numb1 / numb2)
  end
  
  def calcular()
    if operacion == "suma"
      puts suma()
    elsif operacion == "resta"
      puts resta()
    elsif operacion == "multiplicacion"
      puts multiplicacion()
    elsif operacion == "division"
      puts division()
    end
  end


end


puts "Ingrese el primer numero:"
input1 = gets.chomp.to_i
puts "Ingrese el segundo numero:"
input2 = gets.chomp.to_i
puts "Que quiere hacer?: suma, resta, multiplicacion, division"
input3 = gets.chomp
nueva = Calculadora.new(input1,input2,input3)
nueva.calcular

File: keiner_lindarte_aro/instructor_henry/mis_programas/dart/orientado_a_objetos/3_clase_de_02_agosto/actividad.dart
import 'dart:io';

class Calculadora {

  double numb1 = 0;
  double numb2 = 0;
  String operacion = "";

  Calculadora (double n1,double n2,String operacion){

  this.numb1 = n1;
  this.numb2 = n2;
  this.operacion = operacion;

  }

  double suma(){
  return (numb1 + numb2);
  }

  double resta(){
  return (numb1 - numb2);
  }

  double multiplicacion(){
  return (numb1 * numb2);
  }

  double division(){
  return (numb1 / numb2);
  }

  void calcular(){
    if (operacion == "suma"){
      print(suma());
    } else if (operacion == "resta"){
      print(resta());
    } else if (operacion == "multiplicacion"){
      print(multiplicacion());
    } else if (operacion == "division"){
      print(division());
    } else {
      print("No se pudo realizar la operacion");
    }
  }

}

void main (){

  print("Ingrese el primer numero");
  double input1 = double.parse(stdin.readLineSync()!);
  print("Ingrese el segundo numero");
  double input2 = double.parse(stdin.readLineSync()!);
  print("Que quiere hacer? suma, resta, multiplicacion, division");
  String input3 = stdin.readLineSync()!;

  Calculadora nuevo = new Calculadora(input1, input2, input3);
  nuevo.calcular();
  
}